packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -0,0 +1,308 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/scheduler.dart';
+
+import 'binding.dart';
+import 'events.dart';
+import 'pointer_router.dart';
+
+class _MouseDetails {
+  _MouseDetails({this.sourceTimeStamp, this.globalPosition});
+
+  /// Recorded timestamp of the source pointer event that last reported its
+  /// position. May be null if the mouse pointer is removed while the mouse
+  /// pointer is being tracked.
+  final Duration sourceTimeStamp;
+
+  /// The pointer's global position when it triggered this event in logical
+  /// pixels. May be null if the mouse pointer is removed while the mouse
+  /// pointer is being tracked.
+  final Offset globalPosition;
+
+  @override
+  int get hashCode {
+    return hashValues(sourceTimeStamp, globalPosition);
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    return sourceTimeStamp == other.sourceTimeStamp
+      && globalPosition == other.globalPosition;
```

Comment:
this will crash if `other` is not the same type

Reply:
you should use the exact pattern we use in other operator== implementations (e.g. you should be checking runtimeType, not using `is`, otherwise `MouseEnterDetails() == MouseMoveDetails()`).